powerspikegg/gateway/src/test/scala/gg/powerspike/MockGrpcServer.scala

```diff
@@ -0,0 +1,28 @@
+package gg.powerspike
+
+import io.grpc.ServerBuilder
+
+/**
+  * Grpc server which implements services necessary to test the gateway
+  */
+class MockGrpcServer {
+
+  private val server = ServerBuilder
+    .forPort(50001)
+    .addService(new MockFetcher)
+    .build()
+
+  def start(): Unit = server.start()
+
+}
+
+/**
+  * Singleton containing an instance of the Mock Grpc server
+  * Allow to share the server between tests (The server is stateless)
+  */
+object MockGrpcServer {
```

Comment:
Avoid using singletons. gRPC test servers are fast enough to startup / teardown in each test case.